### PR TITLE
TP2000-980: Quota definition create

### PR DESCRIPTION
**Pull request by eadpearce**
# TP2000-980: Quota definition create

## Why

* Users need the ability to create new definition periods for an existing quota order number

## What

* Link to create a new definition on the quota order number detail page under the "Actions" list
* Single page form to create a new definition
* Success page

FireShot Capture 064 - Create a new quota definition period - Manage Trade Tariffs - localhost
Screenshot 2023-10-16 at 15 44 22

### CODE REVIEW

**Comment by SteveCN7** [comment]
Couple of things I think we might need to consider/update for both this design and the edit:

1. End dates should be enforced - Definition periods always run for specific periods of time
2. Some help text on what the two volume fields do

File: quotas/forms.py [hunk 1]
```diff
@@ -287,24 +288,38 @@ class Meta:
...
     initial_volume = forms.DecimalField(
         widget=forms.TextInput(),
-        error_messages={"invalid": "Initial volume must be a number"},
+        error_messages={
+            "invalid": "Initial volume must be a number",
+            "required": "Enter the initial volume",
+        },
+    )
+    measurement_unit = forms.ModelChoiceField(
+        queryset=MeasurementUnit.objects.all(),
```
Comment (a-gleeson) [inline quotas/forms.py:304]: As MeasurementUnit is a trcaked model would we want to use current() so the drop down only returns the latest version?

File: quotas/forms.py [hunk 2]
```diff
@@ -369,6 +384,137 @@ def init_layout(self):
...
+    )
+    initial_volume = forms.DecimalField(
+        widget=forms.TextInput(),
+        error_messages={
+            "invalid": "Initial volume must be a number",
+            "required": "Enter the initial volume",
+        },
+    )
+    measurement_unit = forms.ModelChoiceField(
+        queryset=MeasurementUnit.objects.all(),
```
Comment (a-gleeson) [inline quotas/forms.py:422]: Ditto comment above

File: quotas/jinja2/includes/quotas/actions.jinja [hunk 3]
```diff
@@ -3,7 +3,10 @@
       <h2 class="govuk-heading-s" id="subsection-title">Actions</h2>
       <ul class="govuk-list govuk-!-font-size-16">
             <li><a class="govuk-link" href="{{ url('quota-definitions', args=[object.sid]) }}">View definition periods</a></li>
```
Comment (a-gleeson) [inline quotas/jinja2/includes/quotas/actions.jinja:5]: should there be a `<br>` after this list element too?

Comment (eadpearce) [reply quotas/jinja2/includes/quotas/actions.jinja:5]: No. The top two items are grouped together because they're both to do with definition periods

**Review by a-gleeson — APPROVED** [review]
LGTM!

**Pull request merged**